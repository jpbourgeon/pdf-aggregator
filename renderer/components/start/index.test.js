const React = require('react');
const ShallowRenderer = require('react-test-renderer/shallow');
const { RenderView } = require('../start');
const { defaultState } = require('./store');

describe('The renderView component of the start page', () => {
  it('should render correctly', () => {
    const props = {
      actions: {
        openDialog: jest.fn(),
        setFolder: jest.fn(),
        handleChange: jest.fn(),
        resetState: jest.fn(),
        isDataValid: jest.fn(),
        submit: jest.fn(),
      },
      aboutActions: {
        open: jest.fn(),
      },
      i18nActions: {
        t9n: jest.fn(),
      },
      classes: {
        root: jest.fn(),
        title: jest.fn(),
        formControl: jest.fn(),
        smallFormControl: jest.fn(),
        button: jest.fn(),
      },
      state: defaultState,
    };
    const renderer = new ShallowRenderer();
    renderer.render(<RenderView {...props} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
