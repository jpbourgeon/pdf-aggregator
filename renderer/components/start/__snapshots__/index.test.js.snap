// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The renderView component of the start page should render correctly 1`] = `
<form
  className={[MockFunction]}
  onSubmit={[Function]}
>
  <WithStyles(Component) />
  <WithStyles(GridWrapper)
    container={true}
    spacing={24}
  >
    <WithStyles(GridWrapper)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        className={[MockFunction]}
        variant="display1"
      >
        Paramètres
        <WithStyles(IconButton)
          onClick={[MockFunction]}
          size="small"
          variant="raised"
        >
          <pure(HelpOutline) />
        </WithStyles(IconButton)>
      </WithStyles(Typography)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(InputLabel)
          htmlFor="input"
          shrink={true}
        >
          Dossier source
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          disabled={true}
          id="input"
          startAdornment={
            <WithStyles(InputAdornment)
              position="start"
            >
              <WithStyles(IconButton)
                aria-label="Choisir le dossier source"
                onClick={[Function]}
              >
                <pure(FolderOpen) />
              </WithStyles(IconButton)>
            </WithStyles(InputAdornment)>
          }
          type="text"
          value=""
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(InputLabel)
          htmlFor="level"
          shrink={true}
        >
          Niveau (0 = racine)
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          className={[MockFunction]}
          id="level"
          onChange={[Function]}
          type="number"
          value={0}
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(InputLabel)
          htmlFor="depth"
          shrink={true}
        >
          Profondeur (0 = illimitée)
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          className={[MockFunction]}
          id="depth"
          onChange={[Function]}
          type="number"
          value={0}
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(InputLabel)
          htmlFor="output"
          shrink={true}
        >
          Dossier cible
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          disabled={true}
          id="output"
          label=""
          startAdornment={
            <WithStyles(InputAdornment)
              position="start"
            >
              <WithStyles(IconButton)
                aria-label="Choisir le dossier cible"
                onClick={[Function]}
              >
                <pure(FolderOpen) />
              </WithStyles(IconButton)>
            </WithStyles(InputAdornment)>
          }
          type="text"
          value=""
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(InputLabel)
          htmlFor="filename"
          shrink={true}
        >
          Nom de(s) fichier(s) cible(s) (options : %dossiersouce%, %dateiso%)
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          id="filename"
          onChange={[Function]}
          type="text"
          value="%dossiersource%_%dateiso%"
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(FormControlLabel)
          control={
            <WithStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          label="Page de couverture"
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className="function () {
        return fn.apply(this, arguments);
      }"
      >
        <WithStyles(InputLabel)
          htmlFor="output"
          shrink={true}
        >
          Logo (Formats acceptés : JPEG, PDF)
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          disabled={true}
          id="logo"
          label=""
          startAdornment={
            <WithStyles(InputAdornment)
              position="start"
            >
              <WithStyles(IconButton)
                aria-label="Choisir un logo"
                onClick={[Function]}
              >
                <pure(FolderOpen) />
              </WithStyles(IconButton)>
            </WithStyles(InputAdornment)>
          }
          type="text"
          value=""
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className="function () {
        return fn.apply(this, arguments);
      }"
      >
        <WithStyles(InputLabel)
          htmlFor="title"
          shrink={true}
        >
          Titre (options : %dossiersource%, %dateiso%, %ligne%)
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          id="title"
          onChange={[Function]}
          type="text"
          value="%dossiersource%"
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className="function () {
        return fn.apply(this, arguments);
      }"
      >
        <WithStyles(InputLabel)
          htmlFor="subtitle"
          shrink={true}
        >
          Sous-titre (options : %dossiersource%, %dateiso%, %ligne%)
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          id="subtitle"
          onChange={[Function]}
          type="text"
          value="%dateiso%"
        />
      </WithStyles(FormControl)>
      <br />
      <WithStyles(FormControl)>
        <WithStyles(FormControlLabel)
          control={
            <WithStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          label="Journal des modifications"
        />
        <WithStyles(FormControlLabel)
          control={
            <WithStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          label="Signets"
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)>
        <WithStyles(FormControlLabel)
          control={
            <WithStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          label="Table des matières"
        />
        <WithStyles(FormControlLabel)
          control={
            <WithStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          label="Numérotation des pages"
        />
      </WithStyles(FormControl)>
      <div>
        <WithStyles(Button)
          className={[MockFunction]}
          color="primary"
          disabled={true}
          type="submit"
          variant="raised"
        >
          Valider
        </WithStyles(Button)>
        <WithStyles(Button)
          className={[MockFunction]}
          color="secondary"
          onClick={[MockFunction]}
          variant="raised"
        >
          Annuler
        </WithStyles(Button)>
      </div>
    </WithStyles(GridWrapper)>
  </WithStyles(GridWrapper)>
</form>
`;
